mysql-test/t/acl_mdl_scalability.test

```diff
@@ -0,0 +1,269 @@
+--remove_file $MYSQL_TMP_DIR/thread_limit
+
+--echo # Precondition check of the system is completed. The system can handle 12K concurrent threads
+
+--source include/count_sessions.inc
+
+--let $debug_point = skip_session_admin_check
+--source include/add_debug_point.inc
+
+CREATE USER satya@'%' IDENTIFIED BY 'hello';
+GRANT ALL on test.* to satya@'%';
+GRANT ALL on mysql.* to satya@'%';
+
+--let $old_max_conn=`SELECT @@global.max_connections`
+SET GLOBAL max_connections = 6000;
+
+--connect (con_flush,localhost,root,,)
+--connect (con0,localhost,root,,)
+
+--let $i = 1
+while ($i <= 200) {
+ --connect (con$i, localhost, root)
+ --inc $i
+}
+
+--let $i = 201
+while ($i <= 700) {
+ --connect (con$i, 127.0.0.1, satya,hello,test)
+ --inc $i
+}
+
+--connection default
+--let $initial_con_count=`SELECT COUNT(*) FROM information_schema.processlist`
+
+--echo # Sleeping 15 seconds for allowing user to connect client for observability
+--echo # we cannot establish connection later as database freezes. Do now if you wish
+--echo # be in mysql-source/bld/mysql-test directory and execute the below command
+--echo # ../runtime_output_directory/mysql -uroot -S ./var/tmp/mysqld.1.sock
+#--sleep 15
+#--echo # sleep over
+
+--echo # connection con0: Hold ACL x lock and let readers wait
+--connection con0
+SET DEBUG_SYNC= 'acl_x_lock SIGNAL x_locked WAIT_FOR all_wait_s_locks_done';
+--send FLUSH PRIVILEGES
+
+--echo # 200 connections try for SHARED ACL lock and go into waiting list
+--echo # because FLUSH PRIVILEGES holds EXCLUSIVE lock
+--disable_query_log
+--let $i = 1
+while ($i <= 200) {
+ --connection con$i
+ SET DEBUG_SYNC = 'now WAIT_FOR x_locked NO_CLEAR_EVENT';
+ SET DEBUG_SYNC = 'acl_s_lock SIGNAL s_locked WAIT_FOR new_xlock NO_CLEAR_EVENT';
+ --send SHOW CREATE USER root@localhost
+ --inc $i
+}
+--enable_query_log
+
+--echo # Connection default. Waiting for 200 connections to reach blocking state
+--connection default
+--let $wait_timeout = 600
+let $wait_condition=
+  SELECT COUNT(*) = 200 FROM information_schema.processlist
+    WHERE state = 'Waiting for acl cache lock' AND
+          info = 'SHOW CREATE USER root@localhost';
+--source include/wait_condition.inc
+
+--echo # Now release X-lock by FLUSH PRIVILEGES. Now All 200 readers get the
+--echo # SHARED lock and they wait on debug sync point after acquiring SHARED
+--echo # ACL MDL lock
+SET DEBUG_SYNC = 'now SIGNAL all_wait_s_locks_done';
+
+--echo # Wait for 200 connection to reach acquired SHARED ACL MDL lock state
+--let $wait_timeout = 600
+let $wait_condition=
+ SELECT COUNT(*) = 200 FROM information_schema.processlist
+   WHERE state = 'debug sync point: acl_s_lock' AND
+         info = 'SHOW CREATE USER root@localhost';
+--source include/wait_condition.inc
+
+--echo # Start 5000 connection will go into ACL cache lock waiting state
+--let $i = 1
+while ($i <= 5000) {
+ --exec_in_background $MYSQL -usatya -phello -h 127.0.0.1 -P 13000 test -e "SELECT 1" 2>&1 > /dev/null
```

Comment:
@satya-bodapati `1300` may still be an incorrect value in general case (even if run in non-parallel mode)
I believe `$MASTER_MYPORT` should be used instead.

Reply:
Ah, this is the reason, it was not working with --parallel. Good find. Fixed